xtask/src/release.rs

```diff
@@ -90,13 +180,53 @@ impl flags::Release {
         Ok(remote)
     }
 
-    /// Get the current version of the crate from the manifest.
-    fn get_version(&self, project_root: &Path) -> Result<String> {
-        let manifest_toml = read_file(project_root.join(&self.name).join("Cargo.toml"))?;
+    /// Get the tag name for this release.
+    fn get_tag(&self) -> String {
+        format!("{}-{}", self.name, self.version)
+    }
+
+    /// Get the current version of the crate at `path` from its manifest.
+    fn get_version(path: &Path) -> Result<Version> {
+        let manifest_toml = read_file(path.join("Cargo.toml"))?;
         let manifest: CargoManifest = from_toml_str(&manifest_toml)?;
 
         Ok(manifest.package.version)
     }
+
+    /// Check if the current version of the crate is published on crates.io.
+    fn is_published(&self) -> Result<bool> {
+        let json: JsonValue =
+            self.client.get(format!("{}/{}/{}", CRATESIO_API, self.name, self.version))?.json()?;
+
+        Ok(json.get("version").is_some())
```

Comment:
I generally prefer only deserializing the relevant part of a JSON response. Just because it seems like a lot of unnecessary extra work to build a `serde_json::Value` if one only ends up accessing one field. If you only want to know whether a field exists (and is non-`null`), you could use `Option<serde::de::IgnoredAny>`:

```suggestion
        use serde::de::IgnoredAny;

        #[derive(Deserialize)]
        struct ApiResponse {
            version: Option<IgnoredAny>,
        }

        let response: ApiResponse =
            self.client.get(format!("{}/{}/{}", CRATESIO_API, self.name, self.version))?.json()?;

        Ok(response.version.is_some())
```